presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2769,6 +2770,61 @@ public void testGroupedExecution()
         }
     }
 
+    @Test
```

Comment:
TBH I don't think Hive specific tests are needed

Reply:
I'd prefer to retain them. They setup the data in partitioning/bucketing shape explicitly.

Reply:
I agree with @findepi. The data setup makes it a valid test for the code change.

Reply:
I don't think we should put core specific tests (in this case particular feature in planner) in connector. This adds confusion to what is tested where. We already try to untangle `AbstractTestQueries` which mix tests for various core and connectors features.

> They setup the data in partitioning/bucketing shape explicitly.

If we want to test table scan properties, we should use (or add if missing) `com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder#tableScan` with required stream/node partitioning in planner tests.

Reply:
I removed Hive tests.

Reply:
@sopel39 : Thank you. I see your point. And I do agree with you that using integration-style test to do unit test is not a good idea.

But my understanding of this test is slightly different -- we are not testing a specific feature, but trying to test whether it could work in a end-to-end environment as expected . And I think it's something expected when doing test right? -- we could have both unit tests that assure each individual component works correctly, and integration test assures all the components can play the orchestra in a harmony way.

What do you think? I don't have strong opinions on this specific case though. We can also do it in the current way (removing this Hive integration test as @shixuan-fan did :) ).

Reply:
> But my understanding of this test is slightly different -- we are not testing a specific feature, but trying to test whether it could work in a end-to-end environment as expected

My worry is that this could lead to tests in connectors for minor planner or core features which we should try to avoid. The whole `AddExchanges` is based plan property processing so we could add similar tests for aggregations, union, join, stream vs node partitioning, sort properties.

> What do you think? I don't have strong opinions on this specific case though

Some integration tests are indeed required, but those should be high-level and not operator specific. For instance, from within connector, one is not guaranteed that engine will always skip the exchange as it could decide otherwise depending on skewness or lower CBO plan.

Reply:
as i said (https://github.com/prestodb/presto/pull/12122#discussion_r247092801) i'd too prefer to retain hive-specific test in addition to more unit-like test without Hive.